pkg/engine/transform/transform.go

```diff
@@ -427,13 +427,13 @@ func (t *Transformer) NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry
 func (t *Transformer) NormalizeResourcesForK8sAgentUpgrade(logger *logrus.Entry, templateMap map[string]interface{}, isMasterManagedDisk bool, agentPoolsToPreserve map[string]bool) error {
 	logger.Infoln(fmt.Sprintf("Running NormalizeResourcesForK8sMasterUpgrade...."))
 	if err := t.NormalizeResourcesForK8sMasterUpgrade(logger, templateMap, isMasterManagedDisk, agentPoolsToPreserve); err != nil {
-		log.Fatalln(err)
+		log.Println(err)
```

Comment:
do we need to print the error before we return it? If we're doing things right the error should get logged later on by the caller.

Reply:
Perhaps printing the error is redundant. I was trying to keep the changes minimal and obvious by just translating `Fatalln` to `Println`.

I'll take a look at removing these and the others you mentioned.